_posts/2020-10-27-scalajs-for-scala-3.md

```diff
@@ -0,0 +1,184 @@
+---
+layout: blog-detail
+post-type: blog
+by: Sébastien Doeraene
+title: "Implementing Scala.js Support for Scala 3"
+---
+
+Today, the JS platform has become an integral part of the language.
+Yet, until last August, support for Scala.js in Scala 3 was close to non-existent.
+A first preview was shipped in Dotty 0.27.0-RC1, with support for the portable subset of Scala and native JS types.
+Since then, support for non-native JS types was added and will ship as part of Scala 3.0.0-M1.
+The only missing feature left is JS exports, which we will implement by the next milestone.
+
+What did it take to add Scala.js support in Scala 3?
+How come we managed to do the necessary work in less than 3 months, whereas Scala.js for Scala 2 took 7 years to mature?
```

Comment:
```suggestion
How come did we manage to do the necessary work in less than 3 months, whereas Scala.js for Scala 2 took 7 years to mature?
```
?

Reply:
@julienrf I think you mean "How did we", not "Home come did we"

Reply:
Hmm, I’m not sure… I’m not an english-native, but “How come we managed to” reads weird to me. Maybe it’s correct?

Reply:
You're right that Seb's original was awkward. I'm suggesting "How did we manage to do the necessary work" (no "come", you left the "come" in your version)